cpp/src/sampling/neighbor_sampling_impl.hpp

```diff
@@ -192,7 +215,7 @@ neighbor_sample_impl(
     if (labels) { (*level_result_label_vectors).push_back(std::move(*labels)); }
 
     ++hop;
-    if (hop < fan_out.size()) {
+    if (hop < (*fan_out).size()) {
```

Comment:
fan_out size will (potentially) vary by type.

Reply:
Right I only added it for it to compile. I will revisit this approach once we lock the API's interface. It is only supporting non heterogeneous type for now